airflow/models.py

```diff
@@ -48,7 +49,7 @@
 from sqlalchemy import case, func, or_, and_
 from sqlalchemy.ext.declarative import declarative_base, declared_attr
 from sqlalchemy.dialects.mysql import LONGTEXT
-from sqlalchemy.orm import relationship, synonym
+from sqlalchemy.orm import reconstructor, relationship, synonym
```

Comment:
Agreed, it's pretty cool! I think at some point it would be nice to have one state of TaskInstance if possible (rather than having half of the properties filled in from the DB but missing the rest like task and vice versa when the task instance is constructed using the construcor), maybe by lazy loading properties (e.g. once a DB column is referenced call refresh_from_db automatically).